tests/unit/test_locales.py

```diff
@@ -13,98 +15,104 @@ class TestLocales(SearxTestCase):
-        #      optimized with first priority on territory and second on language.
-
-        # Assume we have an engine that supports the following locales:
-        locale_tag_list = ['zh-CN', 'zh-HK', 'nl-BE', 'fr-CA']
-
-        # Examples (Assumption A.)
-        # ------------------------
-
-        # A user selects region 'zh-TW' which should end in zh_HK.
-        # hint: CN is 'Hans' and HK ('Hant') fits better to TW ('Hant')
-        self.assertEqual(locales.match_locale('zh-TW', locale_tag_list), 'zh-HK')
-
-        # A user selects only the language 'zh' which should end in CN
-        self.assertEqual(locales.match_locale('zh', locale_tag_list), 'zh-CN')
-
-        # A user selects only the language 'fr' which should end in fr_CA
-        self.assertEqual(locales.match_locale('fr', locale_tag_list), 'fr-CA')
-
-        # The difference in priority on the territory is best shown with a
-        # engine that supports the following locales:
-        locale_tag_list = ['fr-FR', 'fr-CA', 'en-GB', 'nl-BE']
-
-        # A user selects only a language
-        self.assertEqual(locales.match_locale('en', locale_tag_list), 'en-GB')
-
-        # hint: the engine supports fr_FR and fr_CA since no territory is given,
-        # fr_FR takes priority ..
-        self.assertEqual(locales.match_locale('fr', locale_tag_list), 'fr-FR')
-
-        # Examples (Assumption B.)
-        # ------------------------
-
-        #  A user selects region 'fr-BE' which should end in nl-BE
-        self.assertEqual(locales.match_locale('fr-BE', locale_tag_list), 'nl-BE')
-
-        # If the user selects a language and there are two locales like the
-        # following:
-
-        locale_tag_list = ['fr-BE', 'fr-CH']
-
-        # The get_engine_locale selects the locale by looking at the "population
-        # percent" and this percentage has an higher amount in BE (68.%)
-        # compared to CH (21%)
-
-        self.assertEqual(locales.match_locale('fr', locale_tag_list), 'fr-BE')
+    @parameterized.expand(
+        [
+            ('zh-TW', ['zh-HK'], 'zh-HK'),  # A user selects region 'zh-TW' which should end in zh_HK.
+            # hint: CN is 'Hans' and HK ('Hant') fits better to TW ('Hant')
+            ('zh', ['zh-CN'], 'zh-CN'),  # A user selects only the language 'zh' which should end in CN
+            ('fr', ['fr-CA'], 'fr-CA'),  # A user selects only the language 'fr' which should end in fr_CA
+            ('nl', ['nl-BE'], 'nl-BE'),  # A user selects only the language 'fr' which should end in fr_CA
+            # Territory tests
+            ('en', ['en-GB'], 'en-GB'),  # A user selects only a language
+            (
+                'fr',
+                ['fr-FR', 'fr-CA'],
+                'fr-FR',
+            ),  # the engine supports fr_FR and fr_CA since no territory is given, fr_FR takes priority
+        ]
+    )
+    def test_locale_optimized_selected(self, locale: str, locale_list: list[str], expected_locale: str):
```

Comment:
All these tests are the same, but I left the comments for context inline to each test case